### DISCUSSION TITLE
[Chapter 09] postForEntity VS exchange

### DISCUSSION THREAD

**Discussion by JoisFe** [cowaine/spring-boot-for-msa#136] [Chapter 09. Application Integration: REST API]
저는 RestTemplate을 자주 사용하나 getForEntity, postForEntity 등 HTTP 메서드에 따라 메서드가 달라지는게 싫어 exchange를 자주 사용했습니다. (물론 인자로 HTTP 메서드를 명시해야하지만)
그런데 이번 책을 통해 해당 장점 보다 더 큰 장점으로`슈퍼 타입 토큰` 이라 부르는 Parameterized TypeReference 인자 라고 합니다.
만약 응답 객체를 커스텀하여 중첩 클래스 타입을 사용한다면 exchange 메서드가 훨씬 편할 것으로 보입니다.

책의 내용만 본다면 exchange가 완전히 상위호환으로 보이는데 exchange 보다 postForEntity를 사용하는게 더 좋은 케이스가 있냐요?

**Comment by ahngilwoong** [comment]
간단한 상황에 명시적으로 좋다 말곤 없는 것 같습니다..
